File: 1704-determine-if-string-halves-are-alike/halves_are_alike.js
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    const countVowels = (str) => {
        const vowels = new Set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
        let count = 0;

        for (const char of str) {
            if (vowels.has(char)) {
                count++;
            }
        }

        return count;
    };

    const halfLength = s.length / 2;
    const firstHalf = s.slice(0, halfLength);
    const secondHalf = s.slice(halfLength);

    return countVowels(firstHalf) === countVowels(secondHalf);
};

// Example usage:
const inputString = "AbCdEfGh";
const result = halvesAreAlike(inputString);
console.log(result); // Output: true

File: 1704-determine-if-string-halves-are-alike/halves_are_alike.py
def halves_are_alike(self, s: str) -> bool:
    def count_vowels(s):
        vowels = set('aeiouAEIOU')
        return sum(1 for char in s if char in vowels)

    half_length = len(s) // 2
    first_half = s[:half_length]
    second_half = s[half_length:]

    return count_vowels(first_half) == count_vowels(second_half)

# Example usage:
input_string = "AbCdEfGh"
result = halves_are_alike(input_string)
print(result)  # Output: True
